presto-parser/src/main/java/com/facebook/presto/sql/tree/LambdaExpression.java

```diff
@@ -22,27 +22,29 @@
 public class LambdaExpression
         extends Expression
 {
-    private final List<String> arguments;
+    private final List<LambdaArgumentDeclaration> arguments;
     private final Expression body;
+    private final boolean bodyContainsSymbolReferences;
```

Comment:
This seems weird. Symbols are a planner concept, so, ideally, the AST shouldn't have to be aware of them.

Also, it's something that can only be determined at analysis/planning time, not during parsing.